Src/FluentAssertions/Primitives/TimeOnlyAssertions.cs

```diff
@@ -127,6 +128,112 @@ public AndConstraint<TAssertions> Be(TimeOnly? expected, string because = "", pa
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="TimeOnly"/>  is within the specified time
+    /// from the specified <paramref name="nearbyTime"/> value.
+    /// </summary>
+    /// <param name="nearbyTime">
+    /// The expected time to compare the actual value with.
+    /// </param>
+    /// <param name="precision">
+    /// The maximum amount of time which the two values may differ.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> BeCloseTo(TimeOnly nearbyTime, TimeSpan precision, string because = "",
+        params object[] becauseArgs)
+    {
+        if (precision < TimeSpan.Zero)
+        {
+            throw new ArgumentOutOfRangeException(nameof(precision), $"The value of {nameof(precision)} must be non-negative.");
+        }
+
+        if (precision == TimeSpan.MaxValue)
+        {
+            precision = precision.AddTicks(-1);
+        }
+
+        TimeOnly minimumValue = nearbyTime.Add(-precision);
+        TimeOnly maximumValue = nearbyTime.Add(precision.AddTicks(1));
```

Comment:
As the interval for `IsBetween` is left-_closed_ and right-_open_, we should only adjust for the ending time.

```cs
TimeOnly minimumValue = nearbyTime.Add(-precision);
TimeOnly maximumValue = nearbyTime.Add((precision == TimeSpan.MaxValue) ? precision : precision.AddTicks(1));
```

Alternatively we could just copy the implementation of [`IsBetween`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs,208) and adjust it to be closed in both ends.